buildpack/bom.go

```diff
@@ -35,13 +35,20 @@ func (v *defaultBOMValidator) ValidateBOM(bp GroupBuildpack, bom []BOMEntry) ([]
 
 func (v *defaultBOMValidator) validateBOM(bom []BOMEntry) error {
 	if len(bom) > 0 {
-		return errors.New("bom table isn't supported in this buildpack api version. The BOM should be written to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>")
+		v.logger.Warn("BOM table is deprecated in this buildpack api version. The BOM should be written to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.")
```

Comment:
To a point that @ekcasey raised in Working Group - maybe we should only warn if a buildpack provided the old bom and NOT the new bom. 

We'd need to extend the `BOMValidator` interface to be able to be able to tell it if bom files exist...

Reply:
I'm not following your statement. The entire `bom.go` concerns the legacy BOM. The bom _files_ validation happens [here](https://github.com/buildpacks/lifecycle/blob/4a6089b4830b0012452406caed1f89c1dcf9aba8/buildpack/bomfile.go#L104-L110).

Also let's not forget that these validations were explicitly requested (_and agreed upon_), so it's fair to ask for clarification when revising - see [slack thread](https://buildpacks.slack.com/archives/CSFE49VGR/p1636662782083900)